include/simgrid/s4u/actor.hpp

```diff
@@ -39,10 +42,29 @@ namespace s4u {
  */
 XBT_PUBLIC_CLASS Actor {
   friend Comm;
+  friend int ::s4u_actor_runner(int argc, char **argv);
+
   Actor(smx_process_t smx_proc);
 public:
-  Actor(const char*name, s4u::Host *host, int argc, char **argv);
-  Actor(const char*name, s4u::Host *host, int argc, char **argv, double killTime);
+  
+  /** Method to create a new actor, the template parameter must extend the class Actor. In addition  the inheritance must be public.  */
+  template<class ActorSubClass> static ActorSubClass *createActor(const char*name, s4u::Host *host, int argc, char **argv) {
+    return createActor<ActorSubClass>( name, host, argc, argv, -1);
+  };
+  
+  /** Method to create a new actor, the template parameter must extend the class Actor. In addition  the inheritance must be public. */
+  template<class ActorSubClass> static ActorSubClass *createActor(const char*name, s4u::Host *host, int argc, char **argv, double killTime) {
```

Comment:
Do we need some sort of smart pointer in order to correctly manage the lifetime of the Actor?